tests/tests_test_workflow/test_dependency_installer.py

```diff
@@ -60,13 +75,9 @@ def test_install_maven_dependencies_remote(self, mock_request, mock_copyfile, mo
             exist_ok=True
         )
         mock_copyfile.assert_not_called()
-        mock_request.assert_has_calls(
-            [
-                call(
-                    "https://ci.opensearch.org/x/y/builds/maven/org/opensearch/notification/alerting-notification-1.2.0.0.jar",
-                    os.path.join(dependency_installer.maven_local_path, "org", "opensearch", "notification", "alerting-notification-1.2.0.0.jar"),
-                )
-            ]
+        mock_request.assert_called_with(
```

Comment:
When I run on my dev box with pre-commit or running pytest I see errors where `/local` is appended to the path.  Is this a bad setup thing?  Curious in case others have seen it
```
>           raise AssertionError(_error_message()) from cause
E           AssertionError: Expected call: urlretrieve('https://ci.opensearch.org/x/y/builds/maven/org/opensearch/notification/alerting-notification-1.2.0.0.jar', '/home/petern/.m2/repository/org/opensearch/notification/alerting-notification-1.2.0.0.jar')
E           Actual call: urlretrieve('https://ci.opensearch.org/x/y/builds/maven/org/opensearch/notification/alerting-notification-1.2.0.0.jar', '/local/home/petern/.m2/repository/org/opensearch/notification/alerting-notification-1.2.0.0.jar')
```

Reply:
That's the different between `path.realpath()` and not I believe, something with symlinks, don't remember. See if you can fix this by comparing `path.realpaths`?

Reply:
Yup, right on.  I made sure the calling patterns align now and it works properly on my machine.